src/OneWayMirror.Core/OneWayMirror.cs

```diff
@@ -351,18 +356,23 @@ private bool ApplyCommitToWorkspaceCore(CommitRange commitRange)
                 return false;
             }
 
-            var checkinMessage = CreateCheckinMessage(commitRange);
+            var ownerInfo = GetCommitOwnerInfo(commitRange.NewCommit);
+            
+            var checkinMessage = CreateCheckinMessage(commitRange, ownerInfo.UserDisplayName);
             if (_confirmBeforeCheckin && !ConfirmCheckin(newCommit, checkinMessage))
             {
                 return false;
             }
 
-            var checkinParameters = new WorkspaceCheckInParameters(_workspace.GetPendingChangesEnumerable(), checkinMessage);
-            checkinParameters.NoAutoResolve = true;
-
             try
             {
-                _workspace.CheckIn(checkinParameters);
+                _workspace.CheckIn(
+                    _workspace.GetPendingChanges(),
+                    author: ownerInfo.MappedTfsAuthorName,
```

Comment:
This requires that the account running the mirror tool has CheckinOther permission (https://msdn.microsoft.com/en-us/library/c327ca1z.aspx); @jasonmalinowski tells me it already does.